File: script/HttpServer.py
from flask import Flask, request, Response
from elasticsearch import Elasticsearch
import json
 
app = Flask(__name__)
es = Elasticsearch(['http://192.168.1.20:9200/'])
index = 'callserv_call_nuance_en'

@app.route('/result', methods=['POST'])
def my_json():
	print request.json
	asr_res = "@"
	reference = (request.json)["reference"]

	try:
		asr_res = (request.json)["channels"]["channel1"]["transcript"][0]["text"]
		
	except:
		asr_res = "@"

	#query
	body = "{\"query\": {\"match_all\": {\"reference\" : \"%s\"}}}" % reference
	q_res = es.search(index=index, body=json.loads(body))
	print q_res
	wavname =  q_res['_source']['wavname']
	print wavname
	
	#insert
	doc = {
	    'wavname': wavname,
	    'reference': reference,
	    'text': asr_res
	}

	s_res = es.index(index=index, doc_type='data', id=reference, body=doc)
	print (s_res['result'])

	return s_res['result']

if __name__ == '__main__':
	app.run(debug = True, host = '0.0.0.0', port = '6000')

File: script/HttpServer.pl
#!/usr/bin/perl

use strict;
use URI::Escape;
use Try::Tiny;
use Data::Dumper;
use Mojolicious::Lite;
use Search::Elasticsearch;

my $index = 'callserv_call_nuance_en';
my $es = Search::Elasticsearch->new(nodes=>['192.168.1.20:9200'], cxn_pool => 'Sniff');

post '/result' => sub
{
	my $self = shift;
	$self->inactivity_timeout(300);
	my $result = $self->req->json;
	my $reference = $result->{reference};	
	my $text = 'NULL';
	my $length = 0;
	my $error_info = 0;

	try
	{
		$error_info = scalar(@{$result->{channels}->{channel1}->{errors}});
		$text = $result->{channels}->{channel1}->{errors}->[0]->{message};
	}
	catch
	{
		$error_info = scalar(@{$result->{errors}});
		$text = "NULL";
	};

	if($error_info == 0)
	{
		$text = $result->{channels}->{channel1}->{transcript}->[0]->{text};
		$length = $result->{channels}->{channel1}->{statistics}->{audio_length};
	}
	
	my $results = $es->search(index => $index, body => {query => {match => {reference => $reference}}});
	my $wavname = $results->{hits}->{hits}->[0]->{_source}->{wavname};
	my $server = $results->{hits}->{hits}->[0]->{_source}->{server};
	
	print $wavname.":".$reference.":".$text.":".$length."\n" if $text and $wavname;	

	if($wavname)
	{
		$es->index(index => $index,
			    type => 'data',
		 	     id  => $wavname,
		 	    body => {
				wavname => $wavname,
	    	     	      reference => $reference,
				   text => $text,
				 server => $server,
			         length => $length
				}
			);
	}
};
	
app->start;
